src/plots/linePlot.ts

```diff
@@ -425,6 +449,69 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+      let min: number;
+      let max: number;
+      let currentSlope: number;
+      let scaledXAcessor = Plot._scaledAccessor(this.x());
+      let scaledYAcessor = Plot._scaledAccessor(this.y());
+
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
+      filteredIndices.push(indices[0]);
+
+      function belongToCurBucket(i: number) {
+        let p1x = scaledXAcessor(data[indices[i]], indices[i], dataset);
+        let p1y = scaledYAcessor(data[indices[i]], indices[i], dataset);
+        let p2x = scaledXAcessor(data[indices[i + 1]], indices[i + 1], dataset);
+        let p2y = scaledYAcessor(data[indices[i + 1]], indices[i + 1], dataset);
+        if (currentSlope == null) {
+          currentSlope = (Math.floor(p1x) === Math.floor(p2x)) ? Infinity : (p2y - p1y) / (p2x - p1x);
+          min = (currentSlope === Infinity) ? p1y : p1x;
+          max = min;
```

Comment:
Since there is very little context on `min`/`max`, the above two lines are not too understandable